extensions/search-notes/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+  accounts: boolean;
+  folders: boolean;
+}
+
+export default function Command() {
+  const [state, setState] = useState<State>({
+    notes: [],
+  });
+  function parseNotes(result: string) {
+    const lines = result.split("\n");
+
+    const notes: Note[] = [];
+    let lastAccount = "";
+    let lastFolder = "";
+    let lastNote: Note | null = null;
+
+    for (const line of lines) {
+      const [key, ...rest] = line.split(": ");
+      const value = rest.join(": ");
+
+      switch (key) {
+        case "account":
+          lastAccount = value;
+          break;
+        case "folder":
+          lastFolder = value;
+          break;
+        case "note":
+          lastNote = {
+            name: value,
+            date: null,
+          } as Note;
+          break;
+        case "date":
+          if (lastNote) {
+            lastNote.date = parse(value, "dddd, D MMMM YYYY at HH:mm:ss");
+            lastNote.folder = lastFolder;
+            lastNote.account = lastAccount;
+            notes.push(lastNote);
+          }
+          break;
+      }
+    }
+
+    notes.sort((a, b) => (a.date && b.date && a.date < b.date ? 1 : -1));
+    setState({ notes: notes });
+  }
+  async function checkCachedNotes() {
+    const cachedNotes = (await getLocalStorageItem("notes")) as string;
+    if (cachedNotes) {
+      parseNotes(cachedNotes);
+    }
+  }
+  async function fetchItems() {
+    const result = await runAppleScript(
+      'set output to ""\n' +
+        'tell application "Notes"\n' +
+        "repeat with theAccount in every account\n" +
+        "set theAccountName to the name of theAccount\n" +
+        'set output to output & "account: " & theAccountName & "\n"\n' +
+        "repeat with theFolder in every folder in theAccount\n" +
+        "set theFolderName to the name of theFolder\n" +
+        'set output to output & "folder: " & theFolderName & "\n"\n' +
+        "repeat with theNote in every note in theFolder\n" +
+        "set theNoteName to the name of theNote\n" +
+        "set theNoteDate to the modification date of theNote\n" +
+        'set output to output & "note: " & theNoteName & "\n"\n' +
+        'set output to output & "date: " & theNoteDate & "\n"\n' +
+        "-- say theNoteNumber\n" +
+        "-- show (note whose id is theNoteNumber)\n" +
+        "-- show note theNoteName\n" +
+        "end repeat\n" +
+        "end repeat\n" +
+        "end repeat\n" +
+        "end tell\n" +
+        "return output"
+    );
+    parseNotes(result);
+
+    await setLocalStorageItem("notes", result);
+  }
+  async function openNote(number: number) {
+    await closeMainWindow();
+    await runAppleScript(
+      'tell application "Notes" \n' + 'show note "' + state.notes[number].name + '" \n' + "end tell"
+    );
+  }
+
+  useEffect(() => {
+    fetchItems();
+    checkCachedNotes();
+  }, []);
+
+  const preferences: Preferences = getPreferenceValues();
+
+  return (
+    <List isLoading={state.notes.length < 1}>
```

Comment:
```suggestion
    <List isLoading={state.notes.length < 1}>
```
This can cause an error, since if no notes are found, it will never stop loading. If no notes are found, you should return an empty array and stop the loader.